src/devices/SM28BYJ48/SM28BYJ48.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Threading;
+
+namespace Iot.Device.SM28BYJ48
+{
+    /// <summary>
+    /// Can move the motor clockwise and counterclockwise.
+    /// </summary>
+    public enum RotationDirection
+    {
+        Stopped,
+        Clockwise,
+        CounterClockwise
+    }
+
+    /// <summary>
+    /// The 28BYJ-48 motor has 512 full engine rotations to rotate the drive shaft once.
+    /// In half-step mode these are 8 x 512 = 4096 steps for a full rotation, and in full-step mode these are
+    /// 4 x 512 = 2048 steps for a full rotation.
+    /// </summary>
+    public enum StepperType
+    {
+        HalfStep,
+        FullStepSinglePhase,
+        FullStepDualPhase
+    }
+
+    /// <summary>
+    /// This class is for controlling stepper motors that are controlled by a 4 pin controller board.
+    /// </summary>
+    /// <remarks>It is tested and developed using the 28BYJ-48 stepper motor and the ULN2003 driver board.</remarks>
+    public class SM28BYJ48 : IDisposable
```

Comment:
Will this work for any stepper motor? if so I'd suspect you can just call it StepperMotor

Reply:
Renamed